sdk/generic-sdk-core/generic-core-http-okhttp/src/main/java/com/generic/core/http/okhttp/implementation/OkHttpResponse.java

```diff
@@ -5,33 +5,110 @@
 
 import com.generic.core.http.models.HttpRequest;
 import com.generic.core.models.BinaryData;
+import com.generic.core.models.HeaderName;
+import okhttp3.Headers;
 import okhttp3.Response;
 import okhttp3.ResponseBody;
 
+import java.util.function.Function;
+
 /**
- * Default HTTP response for OkHttp.
+ * Base response class for OkHttp with implementations for response metadata.
  */
-public final class OkHttpResponse extends OkHttpResponseBase {
-    // Previously, this was 4096, but it is being changed to 8192 as that more closely aligns to what Netty uses as a
-    // default and will reduce the number of small allocations we'll need to make.
-    private static final int BYTE_BUFFER_CHUNK_SIZE = 8192;
+public class OkHttpResponse<T> implements com.generic.core.http.Response<T> {
+    private static final BinaryData EMPTY_BODY = BinaryData.fromBytes(new byte[0]);
 
+    private final BinaryData body;
+    private final Function<com.generic.core.http.Response<?>, ?> deserializationCallback;
+    private final com.generic.core.models.Headers headers;
```

Comment:
same here (I'll stop mentioning it now - just get rid of all the full-qualifications)

Reply:
We need it in some OkHttp classes as they also have a type named `Response`.

Reply:
But I'll make sure to remove it from public method signatures.

Reply:
I'll do you one better, let's remove fully qualified names for Generic Core types and just use them for OkHttp types when necessary (they're shorter).